gulpfile.js

```diff
@@ -0,0 +1,872 @@
+/*jslint node: true, latedef: nofunc*/
+"use strict";
+
+var fs = require('fs');
+var path = require('path');
+var os = require('os');
+var child_process = require('child_process');
+
+var globby = require('globby');
+var jshint = require('gulp-jshint');
+var async = require('async');
+var rimraf = require('rimraf');
+var stripComments = require('strip-comments');
+var mkdirp = require('mkdirp');
+var eventStream = require('event-stream');
+var gulp = require('gulp');
+var gulpInsert = require('gulp-insert');
+var gulpZip = require('gulp-zip');
+var gulpRename = require('gulp-rename');
+var gulpReplace = require('gulp-replace');
+var Promise = require('bluebird');
+var requirejs = require('requirejs');
+
+var packageJson = require('./package.json');
+
+//Gulp doesn't seem to have a way to get the currently running tasks for setting
+//per-task variables.  We use the command line argument here to detect which task is being run.
+var taskName = process.argv[2];
+var noDevelopmentGallery = taskName === 'release' || taskName === 'makeZipFile';
+var copyUnminified = taskName === 'combine' || taskName === 'default' || taskName === undefined;
+var minifyShaders = taskName === 'minify' || taskName === 'minifyRelease' || taskName === 'release' || taskName === 'makeZipFile';
+
+var version = packageJson.version;
+if (/\.0$/.test(version)) {
+    version = version.substring(0, version.length - 2);
+}
+
+var sourceFiles = ['Source/**/*.js',
+                   '!Source/*.js',
+                   '!Source/Workers/**',
+                   '!Source/ThirdParty/Workers/**',
+                   'Source/Workers/createTaskProcessorWorker.js'];
+
+var buildFiles = ['Specs/**/*.js',
+                  '!Specs/SpecList.js',
+                  'Source/Shaders/**/*.glsl'];
+
+var jsHintFiles = ['Source/**/*.js',
+                   '!Source/Shaders/**',
+                   '!Source/ThirdParty/**',
+                   '!Source/Workers/cesiumWorkerBootstrapper.js',
+                   'Apps/**/*.js',
+                   'Apps/Sandcastle/gallery/*.html',
+                   '!Apps/Sandcastle/ThirdParty/**',
+                   'Specs/**/*.js',
+                   'Tools/buildTasks/**/*.js',
+                   'gulpfile.js'];
+
+var filesToClean = ['Source/Cesium.js',
+                    'Build',
+                    'Instrumented',
+                    'Source/Shaders/**.js',
```

Comment:
Generated .js files from subdirectories of Shaders aren't cleaned.  I assume this should be `/**/*.js` instead?